template/9-design.md

```diff
@@ -2,23 +2,40 @@
 
 ## Frontend
 
-*List the key libraries, languages, components used by the MVP.*
+**Implementation Framework** 
 
-- Dart
-- Flutter
-- Firebase
-- GeoFlutterFirePlus
+The application has been developed with Flutter, which is based on the Dart programming language. Flutter can natively compile on Android and iOS from the same code base, which allows us to launch the application to more users easily. 
+We use the Model View ViewModel (MVVM) architecture to ensure structured management and interactions between the UI and the application logic. We use the Riverpod package, which simplifies the application of the Observer pattern, reducing necessary UI refreshes.
 
-*If applicable, describe essential screens.*
+**Firebase communication** 
 
-- Login
-- Home
-- Profile
-- Post Creation
-- Post Details
-- Challenges
-- Map
+We use Firebase Authentication to handle user authentication. The Firebase Core, Firebase Auth and Google Sign-In Flutter packages control the user authentication flow through a Google Sign-In intent on the login screen. Additionally, we plan to add additional OAuth providers, like Apple, to allow most of the users to choose one of the available options and easily authenticate.
 
+Storing and fetching the user interactions in the application are handled using the Firebase Core and Cloud Firestone Flutter packages.
+
+**User location processing** 
+
+In order to retrieve the user's surrounding posts and display the map with the current user location, we use the GeoLocator Flutter package. This package simplifies access to the user's current location on every platform. Additionally, we use GeoFlutterFire Plus, which simplifies the storage and fetching of Firebase documents based on their location by leveraging Geohashing. 
+
+### Essential screens
+
+**Login page**: First screen that appears when the user is not authenticated. It allows the user to authenticate and access the application.
+
+**Feed**: The feed is the central screen of the application. Shown when starting the application as an authenticated user, it displays the posts that are at less than one hundred meters away from the user. This screen allows them to sort the feed by hottest, top, latest, or even nearest posts. The feed displays, for every post, its title, the first seven lines of its description, its owner, the number of comments, the date of the post along with its upvote score. It also allows the user to upvote or downvote any posts directly from the feed screen.
+
+**Profile page**: The profile screen displays all the information associated to the user, such as his username, display name, number of centauri points, badges, number of followers along with his number of followed users. Moreover, it displays the title and description of all of his posts along with all of his comments. The user is able to delete any post or comment that they have created, from anywhere, directly from the profile page.
```

Comment:
You forgot a few "his".

Reply:
You are completely right.

Fixed in: [0feaae4](https://github.com/ProximaEPFL/prd/pull/8/commits/0feaae467797d23a77e9e98b5e9f2fc171ce4269)

Thank you